In [6]:
!pip install ucimlrepo
import pandas as pd
from ucimlrepo import fetch_ucirepo

adult = fetch_ucirepo(id=2)
feat = adult.data.features.copy()
aim = adult.data.targets.copy()

q1 = feat.columns.size + aim.columns.size
print(f"1) Число столбцов в наборе данных: {q1}")

missing_feat = [[i, int(feat[i].isnull().sum())] for i in feat.columns if feat[i].isnull().sum() > 0]
missing_aim = [[i, int(aim[i].isnull().sum())] for i in aim.columns if aim[i].isnull().sum() > 0]
missing_all = missing_feat + missing_aim
if missing_all:
    print(f"2) Пропуски есть в столбцах: {missing_all}")
else:
    print("2) Пропусков в данных нет")

q3 = len(feat['race'].unique())
print(f"3) Количество уникальных значений в столбце race: {q3}")

q4 = feat['hours-per-week'].median()
print(f"4) Медиана hours-per-week: {q4}")

dat = pd.concat([feat, aim], axis=1)
Mzp = len(dat[(dat['income'] == '>50K') & (dat['sex'] == 'Male')])
Wzp = len(dat[(dat['income'] == '>50K') & (dat['sex'] == 'Female')])
if Mzp > Wzp:
    q5 = "мужчин"
else:
    q5 = "женщин"
print(f"5) С ЗП >50K больше {q5} (мужчин: {Mzp}, женщин: {Wzp})")

print("6) Заполнение пропущенных данных:")
for col in ['workclass', 'occupation', 'native-country']:
    if feat[col].isnull().sum() > 0:
        mode_value = feat[col].mode()[0]
        feat[col] = feat[col].fillna(mode_value)
        print(f"   Столбец '{col}': заполнен значением '{mode_value}'")

1) Число столбцов в наборе данных: 15
2) Пропуски есть в столбцах: [['workclass', 963], ['occupation', 966], ['native-country', 274]]
3) Количество уникальных значений в столбце race: 5
4) Медиана hours-per-week: 40.0
5) С ЗП >50K больше мужчин (мужчин: 6662, женщин: 1179)
6) Заполнение пропущенных данных:
   Столбец 'workclass': заполнен значением 'Private'
   Столбец 'occupation': заполнен значением 'Prof-specialty'
   Столбец 'native-country': заполнен значением 'United-States'


1) Число столбцов в наборе данных: 15
2) Пропуски есть в столбцах: [['workclass', 963], ['occupation', 966], ['native-country', 274]]
3) Количество уникальных значений в столбце race: 5
4) Медиана hours-per-week: 40.0
5) С ЗП >50K больше мужчин (мужчин: 6662, женщин: 1179)
6) Заполнение пропущенных данных:
   Столбец 'workclass': заполнен значением 'Private'
   Столбец 'occupation': заполнен значением 'Prof-specialty'
   Столбец 'native-country': заполнен значением 'United-States'
   Также помимо моды в числовых столбцах можно заменять медианой/средним. Либо использовать более сложные алгоритмы для заполнения пропусков на основе данных этой записи в других столбцах